query/src/main/java/org/infinispan/query/dsl/embedded/impl/BaseJPAFilterIndexingServiceProvider.java

```diff
@@ -65,16 +65,18 @@ public void start() {
    @Override
    public void stop() {
       Collection<FilteringListenerInvocation<?, ?>> invocations = filteringInvocations.values();
-      cacheNotifier.getListenerCollectionForAnnotation(CacheEntryActivated.class).removeAll(invocations);
-      cacheNotifier.getListenerCollectionForAnnotation(CacheEntryCreated.class).removeAll(invocations);
-      cacheNotifier.getListenerCollectionForAnnotation(CacheEntryInvalidated.class).removeAll(invocations);
-      cacheNotifier.getListenerCollectionForAnnotation(CacheEntryLoaded.class).removeAll(invocations);
-      cacheNotifier.getListenerCollectionForAnnotation(CacheEntryModified.class).removeAll(invocations);
-      cacheNotifier.getListenerCollectionForAnnotation(CacheEntryPassivated.class).removeAll(invocations);
-      cacheNotifier.getListenerCollectionForAnnotation(CacheEntryRemoved.class).removeAll(invocations);
-      cacheNotifier.getListenerCollectionForAnnotation(CacheEntryVisited.class).removeAll(invocations);
-      cacheNotifier.getListenerCollectionForAnnotation(CacheEntriesEvicted.class).removeAll(invocations);
-      cacheNotifier.getListenerCollectionForAnnotation(CacheEntryExpired.class).removeAll(invocations);
+      if (cacheNotifier != null) {
```

Comment:
They are started here https://github.com/anistor/infinispan/blob/1c26fe427df0422467dfa8de73929feb1eba2890/core/src/main/java/org/infinispan/notifications/cachelistener/CacheNotifierImpl.java#L247 and stopped here https://github.com/anistor/infinispan/blob/1c26fe427df0422467dfa8de73929feb1eba2890/core/src/main/java/org/infinispan/notifications/cachelistener/CacheNotifierImpl.java#L259

If one of them fails due to dependencies or fails during start it cause a complete failure for the remaining providers which do not get to be injected or started at all. And all of them still get stop() invoked later. So yes, that should be fixed in CacheNotifierImpl rather than BaseJPAFilterIndexingServiceProvider .